apps/user-portal/src/pages/consents.tsx

```diff
@@ -48,32 +48,38 @@ export class ConsentsPage extends React.Component<any, any> {
     }
 
     public componentWillMount() {
-        getConsents(ConsentState.ACTIVE).then((response) => {
-            this.setState({ consents: response });
-        });
+        this.updateConsents(ConsentState.ACTIVE);
     }
 
     public handleConsentTabChange = (e, { activeIndex }) => {
         this.setState({ activeIndex }, () => {
-            let consentState = ConsentState.ACTIVE;
-            switch (activeIndex) {
-                case 0:
-                    consentState = ConsentState.ACTIVE;
-                    break;
-                case 1:
-                    consentState = ConsentState.REVOKED;
-                    break;
-                default:
-                    consentState = ConsentState.ACTIVE;
-                    break;
-            }
+            const consentState: ConsentState = this.getConsentState(activeIndex);
+            this.updateConsents(consentState);
+        });
+    }
 
-            getConsents(consentState).then((response) => {
-                this.setState({ consents: response });
-            });
+    public updateConsents = (state: ConsentState) => {
+        getConsents(state).then((response) => {
+            this.setState({ consents: response });
         });
     }
 
+    public getConsentState = (tabIndex: number) => {
```

Comment:
can't we filter the active one from the API response?

Reply:
I've removed the revoked app listing from the code and left the function as it is to give provision for the expired state in the future.